rust, bond: drop ValueError and use InvalidArgument instead

ffmancera opened the pull request:
The integration test for nmstate-rust have been adjusted.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

cathay4t reviewed (changes requested):
I think a better fix is fix bond code to use existing `ErrorKind::InvalidArgument`

ffmancera committed:
rust, bond: drop ValueError and use InvalidArgument instead

Instead of using the newly introduced ValueError, Nmstate should raise
InvalidArgument error.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

cathay4t commented:
@ffmancera You have failure on rust CI `test_create_bond_without_mode`

ffmancera committed:
rust, bond: raise InvalidArgument when mode is missing

In addition, this patch is checking on miimon and arp-interval
validation.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>
